src/core/time.d

```diff
@@ -2627,103 +2632,82 @@ assert(before + timeElapsed == after).
 
 private:
 
-    // static immutable long _ticksPerSecond;
-
-    unittest
+    static @property long pureTicksPerSecond() @trusted pure nothrow @nogc
     {
-        assert(_ticksPerSecond[_clockIdx]);
+        assert(_ticksPerSecond, "_ticksPerSecond not yet initialized");
+        return _ticksPerSecond;
     }
 
-
-    long _ticks;
-}
-
-// This is supposed to be a static variable in MonoTimeImpl with the static
-// constructor being in there, but https://issues.dlang.org/show_bug.cgi?id=14517
-// prevents that from working. This is the workaround that Steven Schveighoffer
-// came up with. Once issue# 14517 has been fixed, this should be changed so
-// that _ticksPerSecond is a static long in MonoTimeImple rather than a static
-// array outside of it.
-private immutable long[__traits(allMembers, ClockType).length] _ticksPerSecond;
-
-@trusted shared static this()
-{
-    version(Windows)
+    static long getTicksPerSecond() @trusted
     {
-        long ticksPerSecond;
-        if(QueryPerformanceFrequency(&ticksPerSecond) != 0)
+        version(Windows)
         {
-            foreach(i, typeStr; __traits(allMembers, ClockType))
-                _ticksPerSecond[i] = ticksPerSecond;
+            long ticksPerSecond = void;
+            if (QueryPerformanceFrequency(&ticksPerSecond) != 0)
+                return ticksPerSecond;
+            else
```

Comment:
Now that this code is returning rather than just assigning, this `else` is unnecessary.